rispy/parser.py

```diff
@@ -120,182 +104,124 @@ def __init__(
             else self.DEFAULT_DELIMITER_MAPPING
         )
         self.ignore = ignore if ignore is not None else self.DEFAULT_IGNORE
-        self.skip_missing_tags = skip_missing_tags
         self.skip_unknown_tags = skip_unknown_tags
         self.enforce_list_tags = enforce_list_tags
         self.newline = newline if newline is not None else self.DEFAULT_NEWLINE
 
     def parse(self, text: str) -> List[Dict]:
         """Parse RIS string."""
-        lines = text.split(self.newline)
-        return list(self._yield_lines(lines))
+        line_gen = (line for line in text.split(self.newline))
+        return self.parse_lines(line_gen)
 
     def parse_lines(self, lines: Union[TextIO, List[str]]):
         """Parse RIS file line by line."""
         return list(self._yield_lines(lines))
 
-    def _yield_lines(self, lines):
-        self.in_ref = False
-        self.current = {}
-        self.last_tag = None
-        self.line_number = -1
+    def _parse_tag(self, line):
+        return (self.get_tag(line), self.get_content(line))
 
-        for line in lines:
-            self.line_number += 1
+    def _iter_till_start(self, lines):
+        while True:
+            line = next(lines)
+            if line.startswith(self.START_TAG):
+                return {self.mapping[self.START_TAG]: self.get_content(line)}
 
-            if self.line_number == 0:
-                line = self.clean_start(line)
+    def _yield_lines(self, lines):
+        last_tag = None
 
-            line = self.clean_text(line)
+        try:
+            record = self._iter_till_start(lines)
 
-            if not line.strip():
-                continue
+            while True:
+                tag, content = self._parse_tag(next(lines))
 
-            if self.is_tag(line):
-                try:
-                    yield self._parse_tag(line)
-                    self.current = {}
-                    self.in_ref = False
-                    self.last_tag = None
-                except NextLine:
+                if tag == "  ":
```

Comment:
This assumes that all multiline content will be nicely formatted with sufficient whitespace so that the content starts at the same character each line. An example where this does not hold is [this test](https://github.com/MrTango/rispy/blob/main/tests/data/example_utf_chars.ris). It might be true for RIS files found in the wild, though?

Reply:
+1, I run into this condition frequently (see the `example_empty_tag.ris` file I partially copied over from an Embase user).

The [prior implementation](https://github.com/MrTango/rispy/blob/d719b24023d136bcf8aa46ae34a90b3829e02a5c/rispy/parser.py#L164) supported this nicely, and a fix here may also address Peter's ^ comment.